### PR TITLE
Azure DevOps pipeline for releasing the library to nuget.org and github releases

### PR DESCRIPTION
**Pull request by danielmbaluka**
Fixes https://github.com/microsoft/OpenAPI.NET.OData/issues/136

Changes in this PR
---
- Added a new Azure DevOps pipeline for automatically building and releasing the package to nuget.org and github releases. The pipeline is configure to be triggered on github tags
- Disabled [delay signing](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming) as recommended in https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming.
- Generated a new strong name signing key for the library that will be used for both debug and release builds

### CODE REVIEW

**Comment by danielmbaluka** [comment]
The pipeline is currently failing because of a key permission issue
`Failed OperationId: 458d5fc7-d8a6-498d-b50f-653a5077485b, Status: FailDoNotRetry, FilePath:D:\a\1\s\bin\Release\net472\Microsoft.OpenApi.OData.Reader.dll, Error: {"code":"NeoSigningServiceFailedError","details":{"error":"An exception occurred in the NEO StaticSigner service.","exception":"This clientId *** is unauthorized to access MSSharedLibSnKey"},"innerError":null}
`
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=64223&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=094908eb-9b91-5864-e5bc-10e3f5697105

**Comment by baywet** [comment]
The main workflow seems to be failing now because of a relative path issue.
adding `workingDirectory: $(Build.SourcesDirectory)\src\Microsoft.OpenApi.OData.Reader\`
under 
https://github.com/microsoft/OpenAPI.NET.OData/blob/dbcf68683d1e21e00af9bfe5338e74556278419f/.github/ISSUE_TEMPLATE/workflows/azure-pipelines_rolling.yml#L58

should fix that

[more information](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/dotnet-core-cli?view=azure-devops)

File: scripts/ValidateProjectVersionUpdated.ps1 [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# API is case-sensitive
+$packageName = $packageName.ToLower()
+$url = "https://api.nuget.org/v3/registration5-semver1/$packageName/index.json"
+
+# Call the NuGet API for the package and get the current published version.
+$nugetIndex = Invoke-RestMethod -Uri $url -Method Get
+$currentPublishedVersion = $nugetIndex.items[0].upper
+
+# Convert the published version into a number e.g. "1.0.9" -> 109
+[int]$currentPublishedVersionNumber = $currentPublishedVersion -replace "\."
```
Comment (baywet) [inline scripts/ValidateProjectVersionUpdated.ps1:37]: the comparison needs to be member by member.
having a published version of 1.0.100 and a project version of 1.2.0 will fail that test although it is correct

File: .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+     ]
+    SessionTimeout: 20
+
+- task: PublishBuildArtifacts@1
+  displayName: 'Publish Artifact - Nuget Packages'
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)\Nuget'
+    ArtifactName: Nuget
+    
+# - task: NuGetCommand@2
```
Comment (baywet) [inline .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml:245]: these tasks should not be commented out

File: .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml [hunk 3]
```diff
@@ -0,0 +1,259 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+name: $(TeamProject)_$(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+# Trigger is on release branches e.g. release/v1.0.10
+trigger:
+  branches:
+    include:
+      - release/*
```
Comment (baywet) [inline .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml:9]: what about triggering on tags instead? this is what we've been using in kiota/java sdk/go sdk and others, works pretty well and prepares the GitHub release

**Review by baywet — CHANGES_REQUESTED** [review]

File: scripts/ValidateProjectVersionUpdated.ps1 (hunk 1 above)
Comment (danielmbaluka) [reply scripts/ValidateProjectVersionUpdated.ps1:37]: Okay - I am updating version comparison to check member by member.

File: .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml (hunk 3 above)
Comment (danielmbaluka) [reply .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml:9]: @baywet using tags works perfectly as long we use tags for official releases (which seem to be practice in this repo https://github.com/microsoft/OpenAPI.NET.OData/tags). I will update the pipeline to trigger on tags. Thanks you.

File: scripts/ValidateProjectVersionUpdated.ps1 (hunk 1 above)
Comment (baywet) [reply scripts/ValidateProjectVersionUpdated.ps1:37]: one way you could make it easier for you would be to rely on the version class from dotnet https://docs.microsoft.com/en-us/dotnet/api/system.version?view=net-6.0 (you can load dotnet classes in PS scripts)

Comment (danielmbaluka) [reply scripts/ValidateProjectVersionUpdated.ps1]: Good suggestion. The script has been updated to compare the version using `System.Version`

File: .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml (hunk 2 above)
Comment (danielmbaluka) [reply .github/ISSUE_TEMPLATE/workflows/azure-pipelines_release.yml]: Addressed. Thank you.

**Comment by danielmbaluka** [comment]
@xuzhg @darrelmiller I would appreciate your reviews on this PR. The significant change in this PR is using the newly generated SNK key for strong name signing. This replaces the shared "MSSharedLibSnKey" which required us to use do delay signing as the private key is only accessible by the Azure pipeline.

**Review by baywet — DISMISSED** [review]

**Review by baywet — APPROVED** [review]

**Review by irvinesunday — APPROVED** [review]

**Pull request merged**